src/component/Calculator.js

```diff
@@ -0,0 +1,24 @@
+import { useState } from "react";
+import CalculatorTotal from "./CalculatorTotal";
+import CalculatorDigit from "./CalculatorDigit";
+import CalculatorModifier from "./CalculatorModifier";
+import CalculatorOperation from "./CalculatorOperation";
+
+const Calculator = () => {
+  const [total, setTotal] = useState("0");
+  const [count, setCount] = useState(0);
+
+  return (
+    <div className="calculator">
+      <CalculatorTotal total={total} />
+      <CalculatorDigit
+        totalState={{ total, setTotal }}
+        countState={{ count, setCount }}
+      />
+      <CalculatorModifier setTotal={setTotal} />
+      <CalculatorOperation totalState={{ total, setTotal }} />
```

Comment:
상태를 최상위 컴포넌트(Calculator)에 선언하고, 
Dispatcher(setTotal, setCount)를 모두 하위 컴포넌트에 그대로 전달하셨네요. 

너무 많은 제어권을 하위 컴포넌트에 전달하면, 상태가 왜, 어떻게 변했는지 쫓아가기가 힘들어 져요 😭

Reply:
그러면 이런 상황을 해결하기 위해서 제가 생각나는 건 context api인데 이것을 사용하는게 맞을까요!?
```suggestion
  const [total, setTotal] = useState("0");

  return (
    <ContextTotal.Provider value={{ total, setTotal }}>
      <div className="calculator">
        <CalculatorTotal />
        <CalculatorDigit />
        <CalculatorModifier />
        <CalculatorOperation />
      </div>
    </ContextTotal.Provider>
  );
```